cloudinit/sources/DataSourceAzure.py

```diff
@@ -979,7 +979,7 @@ def _wait_for_pps_os_disk_shutdown(self):
         raise BrokenAzureDataSource("Shutdown failure for PPS disk.")
 
     @azure_ds_telemetry_reporter
-    def _check_if_nic_is_primary(self, ifname):
+    def _check_if_nic_is_primary(self, ifname: str) -> bool:
         """Check if a given interface is the primary nic or not. If it is the
         primary nic, then we also get the expected total nic count from IMDS.
```

Comment:
update comment - the code is no longer getting the total nic count

Reply:
done!